src/vs/workbench/parts/terminal/electron-browser/terminalLinkHandler.ts

```diff
@@ -129,33 +128,42 @@ export class TerminalLinkHandler {
 		});
 	}
 
-	private _resolvePath(link: string): TPromise<string> {
+	protected _preprocessPath(link: string): string {
 		if (this._platform === platform.Platform.Windows) {
 			// Resolve ~ -> %HOMEDRIVE%\%HOMEPATH%
 			if (link.charAt(0) === '~') {
 				if (!process.env.HOMEDRIVE || !process.env.HOMEPATH) {
-					return TPromise.as(void 0);
+					return null;
 				}
 				link = `${process.env.HOMEDRIVE}\\${process.env.HOMEPATH + link.substring(1)}`;
 			}
-		} else {
-			// Resolve workspace path . / .. -> <path>/. / <path/..
-			if (link.charAt(0) === '.') {
+
+			//resolve relative paths
+			if (!link.match('^' + winDrivePrefix)) {
 				if (!this._contextService.hasWorkspace) {
 					// Abort if no workspace is open
-					return TPromise.as(void 0);
+					return null;
 				}
 				link = path.join(this._contextService.getWorkspace().resource.fsPath, link);
 			}
 		}
-		// Resolve workspace path . / .. -> <path>/. / <path/..
-		if (link.charAt(0) === '.') {
+		// Resolve workspace path . | .. | <relative_path> -> <path>/. | <path>/.. | <path>/<relative_path>
+		else if (link.charAt(0) !== '/' && link.charAt(0) !== '~') {
 			if (!this._contextService.hasWorkspace) {
 				// Abort if no workspace is open
-				return TPromise.as(void 0);
+				return null;
 			}
 			link = path.join(this._contextService.getWorkspace().resource.fsPath, link);
 		}
+		return link;
+	}
+
+	private _resolvePath(link: string): TPromise<string> {
+		link = this._preprocessPath(link);
```

Comment:
Mostly to to test link preprocessing (_preprocessPath() method) in isolation, and not have to worry about it returning a promise in the tests.